java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/jtreg/ActionProviderImpl.java

```diff
@@ -409,12 +409,15 @@ private static boolean newStyleXPatch(FileObject testFile) {
     static String builtClassesDirsForBootClassPath(FileObject testFile) {
         File buildDir = BuildUtils.getBuildTargetDir(testFile);
         Project prj = FileOwnerQuery.getOwner(testFile);
+        Settings settings = prj.getLookup().lookup(Settings.class);
+        boolean useLangtoolsBuild = settings == null || settings.isUseAntBuild();
         List<FileObject> roots = new ArrayList<>();
 
         if (buildDir != null) {
             FileObject repo = prj.getProjectDirectory().getParent().getParent();
             if (repo.getNameExt().equals("langtools") &&
-                ShortcutUtils.getDefault().shouldUseCustomTest(repo.getNameExt(), FileUtil.getRelativePath(repo, testFile))) {
+                ShortcutUtils.getDefault().shouldUseCustomTest(repo.getNameExt(), FileUtil.getRelativePath(repo, testFile)) &&
```

Comment:
Nitpick: maybe a small helper `shouldUseLangtoolsBuild(String repoName, FileObject testFile)` could help to unify this settings lookup + condition: 2 occurrences. Consider when chaning the conditions next time :)